jobserver/models.py

```diff
@@ -340,7 +340,16 @@ def started_at(self):
 
     @property
     def status(self):
-        statuses = self.jobs.values_list("status", flat=True)
+        prefetched_jobs = (
+            hasattr(self, "_prefetched_objects_cache")
+            and "jobs" in self._prefetched_objects_cache
+        )
+        if prefetched_jobs:
+            # make use of prefetch Jobs if we can, this is particularly useful
+            # when filtering JobRequests on the event log page
+            statuses = [j.status for j in self.jobs.all()]
+        else:
+            statuses = self.jobs.values_list("status", flat=True)
```

Comment:
Just for my own understanding, when would we expect this branch to be hit? And if it's there as a "just in case" condition should we consider make it an error to hit this branch so we catch it in testing?

Reply:
Lots of places, anywhere which uses the status icons (there's at least one API using it).  It's definitely a "just in case" but coming from knowing this is the default option for all the places I can't think of!